modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_items.dm

```diff
@@ -0,0 +1,143 @@
+
+/obj/item/soulcatcher_item/attack_self(mob/user, modifiers)
+	linked_soulcatcher.ui_interact(user)
+
+/obj/item/soulcatcher_item/New(loc, ...)
+	. = ..()
+	linked_soulcatcher = AddComponent(/datum/component/soulcatcher)
+
+/obj/item/soulcatcher_item/Destroy(force)
+	if(linked_soulcatcher)
+		qdel(linked_soulcatcher)
+
+	return ..()
+
+/obj/item/soulcatcher_item/attack(mob/living/target_mob, mob/living/user, params)
+	if(!target_mob)
+		return ..()
+
+	var/datum/component/previous_body/body_component = target_mob.GetComponent(/datum/component/previous_body)
+	if(body_component)
+		scan_body(body_component, user)
+		return TRUE
+
+	if(!target_mob.mind || !target_mob.ckey)
+		to_chat(user, span_warning("You are unable to remove a mind from an empty body."))
+		return FALSE
+
+	if(target_mob.stat == DEAD) //We can temporarily store souls of dead mobs.
+		target_mob.ghostize(TRUE) //Incase they are staying in the body.
+		var/mob/dead/observer/target_ghost = target_mob.get_ghost(TRUE, TRUE)
+		if(!target_ghost)
+			to_chat(user, span_warning("You are unable to get the soul of [target_mob]!"))
+			return FALSE
+
+		var/datum/soulcatcher_room/target_room = tgui_input_list(user, "Chose a room to send [target_mob]'s soul to.", name, linked_soulcatcher.soulcatcher_rooms)
+		if(!target_room)
+			return FALSE
+
+		if(tgui_alert(target_ghost, "[user] wants to transfer you to a soulcatcher, do you accept?", name, list("Yes", "No"), autofocus = TRUE) != "Yes")
+			to_chat(user, span_warning("[target_mob] doesn't seem to want to enter."))
+			return FALSE
+
+		if(!target_room.add_soul_from_ghost(target_ghost))
+			return FALSE
+
+		body_component = target_mob.GetComponent(/datum/component/previous_body)
+		if(!body_component)
+			return FALSE
+
+		log_admin("[user] used [src] to put [target_mob]'s mind into a soulcatcher.")
+		scan_body(body_component, user)
+		return TRUE
+
+	var/datum/soulcatcher_room/target_room = tgui_input_list(user, "Chose a room to send [target_mob]'s soul to.", name, linked_soulcatcher.soulcatcher_rooms)
+	if(!target_room)
+		return FALSE
+
+	if(tgui_alert(target_mob, "Do you wish to enter [target_room]?", name, list("Yes", "No")) != "Yes")
+		to_chat(user, span_warning("[target_mob] doesn't seem to want to enter."))
+		return FALSE
+
+	target_room.add_soul(target_mob.mind, TRUE)
+	playsound(src, 'modular_skyrat/modules/modular_implants/sounds/default_good.ogg', 50, FALSE, ignore_walls = FALSE)
+	visible_message(span_notice("[src] beeps: [target_mob]'s mind transfer is now complete."))
+
+	body_component = target_mob.GetComponent(/datum/component/previous_body)
+	scan_body(body_component, user)
+	log_admin("[user] used [src] to put [target_mob]'s mind into a soulcatcher while [target_mob] was still alive.")
+
+	return TRUE
+
+/obj/item/soulcatcher_item/attack_secondary(mob/living/target_mob, mob/living/user, params)
+	if(!target_mob)
+		return FALSE
+
+	if(target_mob.mind || target_mob.ckey || !ishuman(target_mob))
```

Comment:
Does this mean that you can actually transfer it to a mob that's alive?

Reply:
Yes, but because it checks for a client and requires that client to be a human. You would most likely only be able to transfer it to something like a humonkey.